src/Transports.AspNetCore/AuthorizationParameters.cs

```diff
@@ -0,0 +1,68 @@
+#nullable enable
+
+using System.Security.Claims;
+using System.Security.Principal;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+
+namespace GraphQL.Server.Transports.AspNetCore;
+
+/// <summary>
+/// Authorization parameters
+/// </summary>
+public struct AuthorizationParameters<T>
```

Comment:
Mutable struct. Are you sure?

Reply:
I see that all properties are set only in struct ctor. You may add `readonly` to struct and remove all setters.

Reply:
It doesn't matter.  It's used for a single call to `AuthorizationHelper.AuthorizeAsync` and that's it.  Since structs cannot eliminate the public constructor, I usually make them mutable, and it allows to construct the struct either via the constructor or setting the properties.  Mutable structs might allow for adding additional properties in the future without breaking binary compatibility.  Maybe some other variant - `ref struct` or whatever - could be more optimal; I just used a struct so as to (1) eliminate a bunch of parameters to the `AuthorizeAsync` call (2) allow for added properties without breaking compatibility (3) no allocation over class.

Whatever you think is fine...

Reply:
No silver bullet here but I fell into bad situations with mutable structs. 1,2,3 are understandable reasons. 1 and 3 work for readonly struct too. Regarding 2 I don't know. Maybe make it readonly for now and keep in mind that we _can_ change it later in some next major?

Reply:
Ok ... I would like to note that I'm already running into an issue that my authentication library cannot select a specific authentication scheme when multiple schemes are in use with ASP.NET Core.  For example when there is cookie authentication for a MVC app, but a JWT-based GraphQL API is added on top.  I'm still figuring out the proper solution.

Reply:
I am not a specialist in auth stuff.

Reply:
done